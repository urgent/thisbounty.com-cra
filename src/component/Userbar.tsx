import React, { ReactChildren } from 'react'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faUser } from '@fortawesome/free-solid-svg-icons/faUser'

const Container = styled.div`
  display: flex;
`

const Icon = styled(FontAwesomeIcon)`
  stroke: #fff;
  stroke-width: 28px;
  color: #37f;
  margin-left: 4px;
  z-index: 100;
  font-size: 24px;
`

const Bar = styled.svg`
  position: relative;
  left: -8px;
  top: 14px;
`

export function Userbar ({
  users,
  max,
  children
}: {
  users: number
  max: number
  children?: ReactChildren
}) {
  return (
    <Container>
      <Icon icon={faUser} />

      <Bar
        width='170'
        height='37'
        fill='none'
        version='1.1'
        viewBox='0 0 180 37'
        className='sc-hKgILt kWdQcq'
        id='svg131'
      >
        <g
          clip-path='url(#clip0)'
          id='g121'
          transform='translate(-12,-13.088235)'
        >
          <g id='g119'>
            <path
              fill='url(#paint1_linear)'
              d={`m 14,14 h ${(users / max) *
                15 *
                max} c 5.059,-0.106 4.081,8.083 -0.793,7 H 14 Z`}
              id='path115'
              style={{ fill: 'url(#paint1_linear)' }}
            />
            <path
              stroke='#ffffff'
              stroke-width='2'
              d={`m 14,13 h -1 v 9 h ${15 * max - 0.5} a 4.5,4.5 0 0 0 0,-9 z`}
              id='path117'
            />
          </g>
        </g>
        <defs id='defs935'>
          <linearGradient
            id='paint1_linear'
            x1='14'
            x2='178'
            y1='21'
            y2='21'
            gradientUnits='userSpaceOnUse'
          >
            <stop id='stop927' stop-color='#0000FF' />
            <stop id='stop929' offset='1' stop-color='#3377FF' />
          </linearGradient>
          <clipPath id='clip0'>
            <path id='path932' fill='#fff' d='M0 0h188v37H0V0z' />
          </clipPath>
        </defs>
      </Bar>

      {children}
    </Container>
  )
}
