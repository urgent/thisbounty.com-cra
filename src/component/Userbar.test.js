import renderer from 'react-test-renderer';
import React from 'react'
import { Userbar } from './Userbar'

test('Userbar renders correctly', () => {
    const empty = renderer
        .create(<Userbar users={0} max={100} />)
        .toJSON();
    const filled = renderer
        .create(<Userbar users={100} max={100} />)
        .toJSON();
    const tiny = renderer
        .create(<Userbar users={8} max={10} />)
        .toJSON();
    expect(empty).toMatchSnapshot();
    expect(filled).toMatchSnapshot();
    expect(tiny).toMatchSnapshot();
});