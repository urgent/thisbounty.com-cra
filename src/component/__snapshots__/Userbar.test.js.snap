// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Userbar renders correctly 1`] = `
<div
  className="sc-bdfBwQ gbvaeM"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-user fa-w-14 sc-gsTCUz djKOXD"
    data-icon="user"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    className="sc-dlfnbm csLZvP sc-hKgILt kWdQcq"
    fill="none"
    height="37"
    id="svg131"
    version="1.1"
    viewBox="0 0 180 37"
    width="170"
  >
    <g
      clipPath="url(#clip0)"
      id="g121"
      transform="translate(-12,-13.088235)"
    >
      <g
        id="g119"
      >
        <path
          d="m 14,14 h 0 c 5.059,-0.106 4.081,8.083 -0.793,7 H 14 Z"
          fill="url(#paint1_linear)"
          id="path115"
          style={
            Object {
              "fill": "url(#paint1_linear)",
            }
          }
        />
        <path
          d="m 14,13 h -1 v 9 h 1399.5 a 4.5,4.5 0 0 0 0,-9 z"
          id="path117"
          stroke="#ffffff"
          strokeWidth="2"
        />
      </g>
    </g>
    <defs
      id="defs935"
    >
      <linearGradient
        gradientUnits="userSpaceOnUse"
        id="paint1_linear"
        x1="14"
        x2="178"
        y1="21"
        y2="21"
      >
        <stop
          id="stop927"
          stopColor="#0000FF"
        />
        <stop
          id="stop929"
          offset="1"
          stopColor="#3377FF"
        />
      </linearGradient>
      <clipPath
        id="clip0"
      >
        <path
          d="M0 0h188v37H0V0z"
          fill="#fff"
          id="path932"
        />
      </clipPath>
    </defs>
  </svg>
</div>
`;

exports[`Userbar renders correctly 2`] = `
<div
  className="sc-bdfBwQ gbvaeM"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-user fa-w-14 sc-gsTCUz djKOXD"
    data-icon="user"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    className="sc-dlfnbm csLZvP sc-hKgILt kWdQcq"
    fill="none"
    height="37"
    id="svg131"
    version="1.1"
    viewBox="0 0 180 37"
    width="170"
  >
    <g
      clipPath="url(#clip0)"
      id="g121"
      transform="translate(-12,-13.088235)"
    >
      <g
        id="g119"
      >
        <path
          d="m 14,14 h 1400 c 5.059,-0.106 4.081,8.083 -0.793,7 H 14 Z"
          fill="url(#paint1_linear)"
          id="path115"
          style={
            Object {
              "fill": "url(#paint1_linear)",
            }
          }
        />
        <path
          d="m 14,13 h -1 v 9 h 1399.5 a 4.5,4.5 0 0 0 0,-9 z"
          id="path117"
          stroke="#ffffff"
          strokeWidth="2"
        />
      </g>
    </g>
    <defs
      id="defs935"
    >
      <linearGradient
        gradientUnits="userSpaceOnUse"
        id="paint1_linear"
        x1="14"
        x2="178"
        y1="21"
        y2="21"
      >
        <stop
          id="stop927"
          stopColor="#0000FF"
        />
        <stop
          id="stop929"
          offset="1"
          stopColor="#3377FF"
        />
      </linearGradient>
      <clipPath
        id="clip0"
      >
        <path
          d="M0 0h188v37H0V0z"
          fill="#fff"
          id="path932"
        />
      </clipPath>
    </defs>
  </svg>
</div>
`;

exports[`Userbar renders correctly 3`] = `
<div
  className="sc-bdfBwQ gbvaeM"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-user fa-w-14 sc-gsTCUz djKOXD"
    data-icon="user"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    className="sc-dlfnbm csLZvP sc-hKgILt kWdQcq"
    fill="none"
    height="37"
    id="svg131"
    version="1.1"
    viewBox="0 0 180 37"
    width="170"
  >
    <g
      clipPath="url(#clip0)"
      id="g121"
      transform="translate(-12,-13.088235)"
    >
      <g
        id="g119"
      >
        <path
          d="m 14,14 h 112.00000000000001 c 5.059,-0.106 4.081,8.083 -0.793,7 H 14 Z"
          fill="url(#paint1_linear)"
          id="path115"
          style={
            Object {
              "fill": "url(#paint1_linear)",
            }
          }
        />
        <path
          d="m 14,13 h -1 v 9 h 139.5 a 4.5,4.5 0 0 0 0,-9 z"
          id="path117"
          stroke="#ffffff"
          strokeWidth="2"
        />
      </g>
    </g>
    <defs
      id="defs935"
    >
      <linearGradient
        gradientUnits="userSpaceOnUse"
        id="paint1_linear"
        x1="14"
        x2="178"
        y1="21"
        y2="21"
      >
        <stop
          id="stop927"
          stopColor="#0000FF"
        />
        <stop
          id="stop929"
          offset="1"
          stopColor="#3377FF"
        />
      </linearGradient>
      <clipPath
        id="clip0"
      >
        <path
          d="M0 0h188v37H0V0z"
          fill="#fff"
          id="path932"
        />
      </clipPath>
    </defs>
  </svg>
</div>
`;
